src/main/java/com/epam/ta/reportportal/core/configs/rabbit/ReportingConfiguration.java

```diff
@@ -178,38 +177,25 @@ public Queue queueDlq(AmqpAdmin amqpAdmin) {
 	@Bean
 	@Qualifier("reportingListenerContainers")
 	public List<AbstractMessageListenerContainer> listenerContainers(ConnectionFactory connectionFactory,
-			@Qualifier("queues") List<Queue> queues) {
+			ApplicationEventPublisher applicationEventPublisher, @Qualifier("queues") List<Queue> queues) {
 		List<AbstractMessageListenerContainer> containers = new ArrayList<>();
-		Channel channel = connectionFactory.createConnection().createChannel(false);
-		int myQueues = 0;
-		for (Queue queue : queues) {
-			try {
-				if (myQueues < queuesPerPod && getQueueConsumerCount(channel, queue) == 0) {
-					SimpleMessageListenerContainer listenerContainer = new SimpleMessageListenerContainer(connectionFactory);
-					containers.add(listenerContainer);
-					listenerContainer.setConnectionFactory(connectionFactory);
-					listenerContainer.addQueueNames(queue.getName());
-					listenerContainer.setExclusive(true);
-					listenerContainer.setMissingQueuesFatal(false);
-					listenerContainer.setupMessageListener(reportingListener());
-					listenerContainer.afterPropertiesSet();
-					myQueues++;
-					logger.info("Consumer connected to queue {}, myQueues current count is {}", queue.getName(), myQueues);
-				}
-			} catch (Exception e) {
-				logger.error("Trying to connect to queue {}, myQueues current count is {}, exception ", queue.getName(), myQueues, e);
-			}
-		}
-		if (containers.size() < queuesPerPod) {
-			logger.error("Started amount of consumers less then configured (or single node start) : {} < {}", containers.size(), queuesPerPod);
+		int consumersCount = 0;
+		while (consumersCount < queuesPerPod) {
+			SimpleMessageListenerContainer listenerContainer = new SimpleMessageListenerContainer(connectionFactory);
+			containers.add(listenerContainer);
+			listenerContainer.setConnectionFactory(connectionFactory);
+			listenerContainer.addQueues(queues.get(consumersCount));
```

Comment:
don't know for sure, but I have read that it's needed to use addQueueNames() instead of addQueues() here, otherwise some initialization problems possible